scratch.py

```diff
@@ -115,80 +117,96 @@ def pack_string(string):
-        try:
-            fh = open(filename, "rb")
-            magic = fh.read(len(GZIP_MAGIC))
-            if magic == GZIP_MAGIC:
-                fh.close()
-                fh = gzip.open(filename, "rb")
-            else:
-                fh.seek(0)
-            magic = fh.read(len(MAGIC))
-            if magic != MAGIC:
-                raise ValueError("invalid .aib file format")
-            version = fh.read(len(FORMAT_VERSION))
-            if version > FORMAT_VERSION:
-                raise ValueError("unrecognized .aib file version")
-            self.clear()
-        except Exception as err:
-            print(err)
 
+        # try:
+        file = open(filename, "rb")
+        magic = file.read(len(GZIP_MAGIC))
+        if magic == GZIP_MAGIC:
+            file.close()
+            file = gzip.open(filename, "rb")
+        else:
+            file.seek(0)
+
+        magic = file.read(len(MAGIC))
+        if magic != MAGIC:
+            raise ValueError("invalid .aib file format")
+
+        version = file.read(len(FORMAT_VERSION))
+        if version > FORMAT_VERSION:
+            raise ValueError("unrecognized .aib file version")
+
+        self.clear()
+        # except Exception as err:
+        #     print(err)
+
+        data = {}
         while True:
-            report_id = unpack_string(fh, False)
-            if report_id is None:
+            report_id = unpack_string(file, False)
+            if report_id is None:  # End of data
                 break
-            data = {}
             data["report_id"] = report_id
-            for name in ("airport", "aircraft_id",
-                         "aircraft_type", "narrative"):
-                data[name] = unpack_string(fh)
-                print(data[name],34)
-                other_data = fh.read(NumbersStruct.size)
+            print(f"\n{report_id= }\n")
+
+            for name in ("airport", "aircraft_id", "aircraft_type", "narrative"):
+                data[name] = unpack_string(file)
+                print(f"{name}= {data[name]!r}\n")
+
+            other_data = file.read(NumbersStruct.size)
             numbers = NumbersStruct.unpack(other_data)
             data["date"] = datetime.date.fromordinal(numbers[0])
             data["pilot_percent_hours_on_type"] = numbers[1]
             data["pilot_total_hours"] = numbers[2]
             data["midair"] = numbers[3]
+
             incident = Incident(**data)
-            self[incident.report_id] = incident
-            print(89)
-        return True
+            self[report_id] = incident
+            data.clear()
+
+
 z = IncidentCollections()
 z[0] = Incident("03434343",datetime.datetime.now(),"heathr","bone", "erereded",90,midair=True,pilot_total_hours=99900)
-z.export_binary(r"C:\Users\DAVID\.PyCharmCE2019.1\config\scratches\me.bin")
-print(z.import_binary(r"C:\Users\DAVID\.PyCharmCE2019.1\config\scratches\me.bin"))
\ No newline at end of file
+z.export_binary("./me.bin")
+z.import_binary("./me.bin")
```

Comment:
The work on any system and creates the file in the CWD... due to the exclusion of '*.bin' files in the .gitignore, the file won't be tracked by git.